File: domain/src/main/java/com/github/gmetal/domain/model/MediaDetail.java
package com.github.gmetal.domain.model;

import java.util.ArrayList;
import java.util.List;
import org.parceler.Parcel;

@Parcel
public class MediaDetail {

    public Boolean adult;
    public String backdropPath;
    public String belongsToCollection;
    public Integer budget;
    public List<String> genres = new ArrayList<>();
    public String homepage;
    public Integer id;
    public String imdbId;
    public String originalLanguage;
    public String originalTitle;
    public String overview;
    public Double popularity;
    public String posterPath;
    public String releaseDate;
    public Integer revenue;
    public Integer runtime;
    public String status;
    public String tagline;
    public String title;
    public Boolean video;
    public Float voteAverage;
    public Integer voteCount;

    public MediaDetail() {

        this.belongsToCollection = "";
        this.budget = 0;
        this.imdbId = "-1";
        this.revenue = 0;
        this.runtime = 0;
        this.tagline = "";
    }
}

File: domain/src/main/java/com/github/gmetal/domain/model/MediaDetail.kt
package com.github.gmetal.domain.model

import com.github.gmetal.domain.model.MediaItem.Companion.IMAGE_PREFIX

const val DEFAULT_ID = -1
fun Int?.orDefaultId(): Int = this ?: DEFAULT_ID
fun String?.imagePrefixPathOrEmpty(): String = if (this != null) IMAGE_PREFIX + this else ""

open class MediaDetail {

    var adult: Boolean? = null
    var backdropPath: String = ""
    var belongsToCollection: MediaCollection = MediaCollection.empty()
    var budget: Int? = null
    var genres: MutableList<String> = mutableListOf()
    var homepage: String = ""
    var id: Int? = null
    var imdbId: String
    var originalLanguage: String = ""
    var originalTitle: String = ""
    var overview: String = ""
    var popularity: Double? = null
    var posterPath: String = ""
    var releaseDate: String = ""
    var revenue: Int? = null
    var runtime: Int? = null
    var status: String = ""
    var tagline: String
    var title: String = ""
    var video: Boolean? = null
    var voteAverage: Float? = null
    var voteCount: Int? = null

    init {
        this.budget = 0
        this.imdbId = "-1"
        this.revenue = 0
        this.runtime = 0
        this.tagline = ""
    }
}
